File: leetcode-04/solutions/python3/0940.py
class Solution:
    def distinctSubseqII(self, S: str) -> int:
        dict = [0] * 26

        for c in S:
            dict[ord(c) - ord('a')] = (sum(dict) + 1) % int(1e9+7)

        return sum(dict) % int(1e9 + 7)

File: leetcode-04/solutions/cpp/0940.cpp
class Solution {
 public:
  int distinctSubseqII(string S) {
    vector<long> map(26);

    for (char c : S)
      map[c - 'a'] = accumulate(map.begin(), map.end(), 1L) % int(1e9 + 7);

    return accumulate(map.begin(), map.end(), 0L) % int(1e9 + 7);
  }
};